code/datums/components/marionette.dm

```diff
@@ -0,0 +1,59 @@
+/**
+ * Marionette component
+ *
+ * Upon being grabbed, we will align the direction of the parent with the direction of the grabber when they rotate.
+ * While grabbed, will also speak out whatever the original person says
+ */
+/datum/component/marionette
+	/// Weakref to a person grabbing the owner of the component, which we align our direction to.
+	var/datum/weakref/grabbing_ref
+
+/datum/component/marionette/Destroy()
+	var/mob/living/grabber = grabbing_ref?.resolve()
+	if(grabbing_ref)
+		UnregisterSignal(grabber, list(COMSIG_MOVABLE_KEYBIND_FACE_DIR, COMSIG_MOB_SAY))
+	grabbing_ref = null
+	return ..()
+
+/datum/component/marionette/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_LIVING_TRYING_TO_PULL, PROC_REF(on_pull))
+	RegisterSignal(parent, COMSIG_ATOM_NO_LONGER_PULLED, PROC_REF(on_stop_pull))
+
+/datum/component/marionette/UnregisterFromParent()
+	UnregisterSignal(parent, list(
+		COMSIG_LIVING_TRYING_TO_PULL,
+		COMSIG_ATOM_NO_LONGER_PULLED,
+	))
+	return ..()
+
+///Called when something starts pulling us, we now listen in to that thing for rotation.
+/datum/component/marionette/proc/on_pull(atom/movable/source, atom/movable/puller, force)
+	SIGNAL_HANDLER
+
+	if(!puller)
+		return
+	grabbing_ref = WEAKREF(puller)
+	RegisterSignal(puller, COMSIG_MOVABLE_KEYBIND_FACE_DIR, PROC_REF(on_puller_turn))
+	RegisterSignal(puller, COMSIG_MOB_SAY, PROC_REF(on_puller_speech))
```

Comment:
Why bother using weakrefs if we're registering signals on it anyways? 
Just use COMSIG_QDELETING instead, given we have a vested interest in tracking whether it's been deleted or not

Reply:
So we can unregister the puller when they let go

Reply:
Yeah, I get that, but I'm saying since we're already registering a few signals on it there's no reason not to opt for a reference + using `COMSIG_QDELETING` vs holding a weakref

Reply:
I don't see the difference in it, the weakref doc says to use it on things that don't "own" a ref and I thought it was the case here.

Reply:
We've got SOME stake in it by nature of registering a bunch of signals on it

I also just think it's so much cleaner to do 
```dm
thing = new_thing
RegisterSignal(thing, SIGNAL_1, ...)
RegisterSignal(thing, SIGNAL_2, ...)
RegisterSignal(thing, COMSIG_QDELETING, ...)
```
and
```dm
UnregisterSignal(thing, list(SIGNAL_1, SIGNAL_2, COMSIG_QDELETING))
thing = null
```

vs
```dm
thing = WEAKREF(new_thing)
RegisterSignal(new_thing, SIGNAL_1, ...)
RegisterSignal(new_thing, SIGNAL_2, ...)
```
and
```dm
var/atom/maybe_thing = thing?.resolve()
if(maybe_thing)
     UnregisterSignal(maybe_thing, list(SIGNAL_1, SIGNAL_2))
```